content/c/concepts/preprocessors/preprocessors.md

```diff
@@ -0,0 +1,73 @@
+---
+Title: 'Preprocessors'
+Description: 'A preprocessor scans the C program source code before it is compiled, allowing the source code to expand macros, include other files, and conditionally compile blocks of source code.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Command Line'
+  - 'Developer Tools'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+A preprocessor scans the C program source code before it is compiled, allowing the source code to expand macros, include other files, and conditionally compile blocks of source code.
+
+Preprocessor commands and macros are included in the source code file and are interpreted before the source code is passed to the compiler.
+The preprocessor can modify the final code that is compiled or provide messages to the developer during preprocessing.
+
+## Common Uses of Preprocessor Commands
+
+### Including Additional Files
+
+The most common preprocessor command is `#include` which is used to include the text contents of some other file into the one being compiled.
+
+```c
+#include <stdio.h>
+```
+
+The preprocessor will replace `#include <stdio.h>` with the text contents of the file **stdio.h**.
+
+### Conditional Compilation
+
+The preprocessor includes several statements used for conditional compilation: `#if`,`#elif`,`#else` and `#endif`.
+The preprocessor also includes `#ifdef` and `#ifndef` which are shorthand for `#if defined(...)` and `#if !defined(...)`.
+
+```c
+#if HELLO > 0
+  printf("Hello world!");
+#endif
+```
+
+If the macro `HELLO` is greater than zero, then the `printf("Hello world!")` command will be compiled.
+If not, then the command will be omitted from the final code being compiled.
+
+```c
+#ifdef _WIN32
+  // Compile for 32-bit Windows system
+#elif defined(_WIN64)
+  // Compile for a 64-bit Windows system
+#else
+  // Compile for something else
+#endif
+```
+
+The above will compile different code for different operating systems based on if the macros `_WIN32` or `_WIN64` exist.
+
+### Defining Macros
+
+When the preprocessor encounters a macro in the source code, it will replace the macro with the value it has been assigned.
+Macros are defined by using `#define` and can be undefined using `#undef`.
```

Comment:
```suggestion
When the preprocessor encounters a macro in the source code, it will replace the macro with the value it has been assigned. Macros are defined by using `#define` and can be undefined using `#undef`.
```